compiler/src/iree/compiler/Codegen/LLVMGPU/ROCDLConfigureBufferInstructions.cpp

```diff
@@ -121,10 +107,20 @@ struct ROCDLConfigureBufferInstructionsPass final
     if (!clROCDLlEnableBufferInstructions)
       return;
     mlir::FunctionOpInterface funcOp = getOperation();
-    // Is this really he best way to skip this pass on non-rocdl targets?
+    // Is this really the best way to skip this pass on non-rocdl targets?
     IREE::GPU::TargetAttr target = getGPUTargetAttr(funcOp);
     if (!target || !target.isAMD())
       return;
+
+    // Initialize the DataFlowSolver with IntegerRangeAnalysis.
+    DataFlowSolver solver;
+    solver.load<dataflow::DeadCodeAnalysis>();
```

Comment:
I think the usual incantations also want SCCP in here.

... someone really needs to clean that up.